src/ucan-cloud/ucan-gateway.ts

```diff
@@ -0,0 +1,367 @@
+    // Clock stuff
+    const clockStoreName = `fireproof/${dbName}/clock/delegation`;
+    const clockStore = await stateStore(clockStoreName);
+
+    let clock: Client.Clock;
+
+    const dataExport = await clockStore.load();
+
+    if (dataExport) {
+      const data = AgentData.fromExport(dataExport);
+
+      // data.principal
+      const keys = Array.from(data.delegations.keys());
+      const delegation = data.delegations.get(keys[0])?.delegation;
+
+      if (!delegation) throw new Error("Expected a clock delegation to be present");
+
+      clock = {
+        delegation,
+        did: () => data.principal.did(),
+        signer: () => data.principal,
+      };
+    } else {
+      const audience = Absentee.from({ id: DidMailto.fromEmail(email) });
+
+      clock = await Client.createClock({ audience });
+
+      const raw: AgentDataExport = {
+        meta: { name: clockStoreName, type: "service" },
+        principal: clock.signer().toArchive(),
+        spaces: new Map(),
+        delegations: new Map([exportDelegation(clock.delegation)]),
+      };
+
+      await clockStore.save(raw);
+
+      const registration = await Client.registerClock({ clock, server, service });
+      if (registration.out.error) throw registration.out.error;
+    }
+
+    console.log("⏰ CLOCK", clock.did());
+
+    this.inst = { clock, email, server, service, w3 };
+
+    // Start URI
+    return baseUrl.build().defParam("version", "v0.1-ucan").URI();
+  }
+
+  async close(): Promise<bs.VoidResult> {
+    return Result.Ok(undefined);
+  }
+
+  async destroy(): Promise<Result<void>> {
+    return Result.Ok(undefined);
+  }
+
+  async put(url: URI, body: Uint8Array): Promise<bs.VoidResult> {
+    const result = await exception2Result(() => this.#put(url, body));
+    if (result.isErr()) this.logger.Error().Msg(result.Err().message);
+
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+    if (result.isErr()) console.log("🚨", store, result.Err().constructor.name, result.Err().message);
+
+    return result;
+  }
+
+  async #put(url: URI, body: Uint8Array): Promise<void> {
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+
+    if (this.inst === undefined) {
+      throw new Error("Not started yet");
+    }
+
+    const key = url.getParam("key");
+    if (!key) {
+      throw new Error("Key not found in the URI");
+    }
+
+    const name = url.getParam("name");
+    if (!name) {
+      throw new Error("Name not found in the URI");
+    }
+
+    console.log("🥘 PUT", store, key);
+
+    switch (store.toLowerCase()) {
+      case "data": {
+        await Client.store({
+          agent: this.inst.w3.agent.issuer,
+          bytes: body,
+          cid: CID.parse(key).toV1(),
+          server: this.inst.server,
+          service: this.inst.service,
+        });
+        break;
+      }
+
+      case "meta": {
+        const bodyWithCrypto = await bs.addCryptoKeyToGatewayMetaPayload(url, this.sthis, body);
```

Comment:
the exposure of the key should populated into ucan --- 
we have to discuss how we could make that work for demos